server/lib/nl/common/utils.py

```diff
@@ -104,11 +107,53 @@ def sv_existence_for_places(places: List[str], svs: List[str],
   return existing_svs, existsv2places
 
 
+def _facet_contains_date(facet_data, facet_metadata, date) -> bool:
+  if not date:
+    # if there is no date specified, then consider this facet has the date
+    return True
+  date_string = server.lib.nl.detection.date.get_date_string(date)
+  facet_earliest_date = facet_data.get('earliestDate', '')
+  facet_latest_date = facet_data.get('latestDate', '')
+  # If the facet dates are different granularity, consider facet to not have
+  # the date.
+  # TODO: allow for different granularity but same year.
+  if (len(date_string) != len(facet_earliest_date) or
+      len(date_string) != len(facet_latest_date)):
+    return False
+  # If the specified date is outside the range of dates of
+  # this facet, move on to next facet
+  if (date_string < facet_earliest_date or date_string > facet_latest_date):
+    return False
+  # Use observation period of the facet to predict if facet has data for
+  # date of interest.
+  # TODO: this is not 100% accurate, so frontend needs to fail gracefully
+  # when a tile doesn't have data
+  obs_period = facet_metadata.get('observationPeriod', '')
```

Comment:
If obs_period is empty, and date_string, facet_earliest_date and facet_latest_date are all YYYY, then return True?

Reply:
Because there are some yearly datasets which do not set obsPeriod.

Reply:
done